src/main/java/frc/utils/AllianceUtilities.java

```diff
@@ -0,0 +1,127 @@
+package frc.utils;
+
+import frc.robot.constants.FieldConstants;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.Timer;
+
+public class AllianceUtilities {
+
+    private static DriverStation.Alliance ALLIANCE = DriverStation.Alliance.Blue;
+
+    private static double BLUE_ALLIANCE_CHECK_TIMESTAMP = -1;
+
+    /**
+     * @return whether the robot is on the blue alliance
+     */
+    public static boolean isBlueAlliance() {
+        final double timestamp = Timer.getFPGATimestamp();
+        if (timestamp - BLUE_ALLIANCE_CHECK_TIMESTAMP > FieldConstants.MINIMUM_TIMESTAMP_ALLIANCE_CHECK) {
```

Comment:
Do the check and update in another function

Reply:
Depricated